argo/providers/vai/provider.go

```diff
@@ -203,116 +218,184 @@ func (vaip VAIProvider) DeletePipeline(ctx context.Context, providerConfig VAIPr
 	return nil
 }
 
-func (vaip VAIProvider) createSchedulerJobPb(providerConfig VAIProviderConfig, runScheduleDefinition RunScheduleDefinition) (*schedulerpb.Job, error) {
-	schedule, err := ParseCron(runScheduleDefinition.Schedule)
-	if err != nil {
-		return nil, err
+func (vaip VAIProvider) CreateRun(ctx context.Context, providerConfig VAIProviderConfig, runDefinition RunDefinition) (string, error) {
+	runId := runDefinition.Name.Name
+	run := VAIRun{
+		RunId:             runId,
+		Labels:            runLabelsFromRunDefinition(runDefinition),
+		PipelineUri:       providerConfig.pipelineUri(runDefinition.PipelineName.Name, runDefinition.PipelineVersion),
+		RuntimeParameters: runDefinition.RuntimeParameters,
+		Artifacts:         runDefinition.Artifacts,
 	}
 
-	runIntent := RunIntent{
-		RunConfigurationName: runScheduleDefinition.RunConfigurationName,
-		PipelineName:         runScheduleDefinition.PipelineName,
-		PipelineVersion:      runScheduleDefinition.PipelineVersion,
-		RuntimeParameters:    runScheduleDefinition.RuntimeParameters,
-		Artifacts:            runScheduleDefinition.Artifacts,
+	return runId, vaip.SubmitRun(ctx, providerConfig, run)
+}
+
+func (vaip VAIProvider) DeleteRun(_ context.Context, _ VAIProviderConfig, _ string) error {
+	return nil
+}
+
+func (vaip VAIProvider) buildPipelineJob(providerConfig VAIProviderConfig, runScheduleDefinition RunScheduleDefinition) (*aiplatformpb.PipelineJob, error) {
+	parameters := make(map[string]*aiplatformpb.Value, len(runScheduleDefinition.RuntimeParameters))
+
+	parameters = pipelines_util.MapValues(runScheduleDefinition.RuntimeParameters, func(value string) *aiplatformpb.Value {
+		return &aiplatformpb.Value{
+			Value: &aiplatformpb.Value_StringValue{
+				StringValue: value,
+			},
+		}
+	})
+
+	// TODO migrate to non deprecated `ParameterValues` rather than `Parameters` below
```

Comment:
Thinking about this TODO more, we should add why we can't do this right now. I.e. it's not a todo but a note

Reply:
Done